### ISSUE TITLE
zip with password bug

### ISSUE THREAD

**Issue by maxchowhk** [gchq/CyberChef#671]
I have used the zip function with a password to encrypt a file,  however, zip succeeded but the password was not matched when decoding.

**Comment by n1474335** [comment]
Can you provide some example data please? This seems to work for me: https://gchq.github.io/CyberChef/#recipe=Zip('file.txt','','secret','Deflate','MSDOS','Dynamic%20Huffman%20Coding')Unzip('secret',false)&input=VGVzdA

**Comment by n1474335** [comment]
Yes, I'm seeing that too. I think this is due to the difference between encrypting each file in the archive separately and encrypting the archive itself. We should make this a user choice.

**Comment by mubix** [comment]
Having the same issue :/

**Comment by slw07g** [comment]
Experienced the same issue. unzipping it may work inside of CyberChef, but when I try to use any other tool to unzip a file that I zipped+encrypted with cyberchef, the unzipping fails. 

For example, try unzipping the file CyberChef generates with Windows built-in unzip and 7-zip.

Interestingly, when I uploaded a file to a sandbox which handles encrypted zips, the sandbox was able to unzip the file with no issue.

**Comment by d-Rickyy-b** [comment]
I also came across this issue. zlib.js seems to be abandoned since it's last change was five years ago.
